### PR TITLE
Partial accept

### PR DESCRIPTION
**Pull request by rixo**
Implement Vite's partial accept, to fix HMR glitches with components that use `context:module` (see  https://github.com/sveltejs/vite-plugin-svelte/issues/134, for example).

Notice that there is really only one change, in `packages/svelte-hmr/lib/make-hot.js`. I should probably not have updated/added the demo apps in the same PR :sweat_smile: And then, there's the required configuration in `playground/vite-app/vite.config.js`.

To appreciate the effect, edit `ModuleContext.svelte` in the Vite app, with or without the partial accept config, and observe the difference in Vite's HMR logs. With partial accept, `ModuleContext.svelte` and `CmpUsingContext.svelte` get updated (but not `CmpUsingOnlyDefault.svelte`) and the change of the export from `module:context` is correctly reflected. Without partial accept, only `ModuleContext.svelte` gets updated (because it eats the HMR bubble), and changing the value of `y` isn't reflected correctly.

### CODE REVIEW

File: playground/kit-app/package.json [hunk 1]
```diff
@@ -1,30 +1,24 @@
...
-    "prettier-plugin-svelte": "^2.4.0",
-    "svelte": "^3.42.6",
-    "svelte-check": "^2.2.6",
-    "svelte-preprocess": "^4.9.4",
-    "tslib": "^2.3.1",
-    "typescript": "^4.4.3"
-  },
-  "type": "module"
+	"name": "kit-app",
+	"version": "0.0.1",
```
Comment (dominikg) [inline playground/kit-app/package.json:4]: must be `"private": true` or changesets would publish. Same for all other playgrounds

**Comment by dominikg** [comment]
Should we enable this by default in v-p-s?

File: playground/kit-app/package.json (hunk 1 above)
Comment (rixo) [reply playground/kit-app/package.json:4]: Oh, right!

**Comment by rixo** [comment]
> Should we enable this by default in v-p-s?

Yes, I think so. This is arguably a bug fix.

**Review by dominikg — APPROVED** [review]

**Pull request merged**